package GCToCS;

enum MsgNum
{
	eMsgToGSToCSFromGC_Begin = 8191;
	//登陆登出公告领奖
	eMsgToGSToCSFromGC_AskLogin = 8193;
	eMsgToGSToCSFromGC_AskReconnectGame = 8194;
	eMsgToGSToCSFromGC_AskComleteUserInfo = 8195;
	eMsgToGSToCSFromGC_AskUserGameInfo = 8196;
	eMsgToGSToCSFromGC_UserAskGetCLReward = 8197;
	//公告相关
	eMsgToGSToCSFromGC_Notice = 8200;
	eMsgToGSToCSFromGC_AskCurtNotice = 8201;
	//好友系统
	eMsgToGSToCSFromGC_AskAddToSNSList = 8210;
	eMsgToGSToCSFromGC_AskRemoveFromSNSList = 8211;
	eMsgToGSToCSFromGC_AskSendMsgToUser = 8212;
	eMsgToGSToCSFromGC_AskInviteFriendsToBattle = 8213;
	eMsgToGSToCSFromGC_AskQueryUserByNickName = 8214;
	eMsgToGSToCSFromGC_GCReplyAddFriendRequst = 8215;
	eMsgToGSToCSFromGC_GCReplyInviteToBattle = 8216;
	eMsgToGSToCSFromGC_AskAddToSNSListByID = 8217;
	eMsgToGSToCSFromGC_AskCanInviteFriends = 8218;
	eMsgToGSToCSFromGC_AskBlackListOnlineInfo = 8502;
	//道具符文系统
	eMsgToGSToCSFromGC_BUYRunes = 8220;
	eMsgToGSToCSFromGC_EuipRunes = 8221;
	eMsgToGSToCSFromGC_UnloadRunes = 8222;
	eMsgToGSToCSFromGC_ComposeRunes = 8223;
	eMsgToGSToCSFromGC_AskBuyGoods = 8224;
	eMsgToGSToCSFromGC_AskRecoinRune = 8503;
	//邮件系统
	eMsgToGSToCSFromGC_AskMailInfo = 8230;
	eMsgToGSToCSFromGC_AskGetMailGift = 8231;
	//新手引导消息
	eMsgToGSToCSFromGC_AskGuideBattle = 8240;
	eMsgToGSToCSFromGC_AskGuideStepId = 8241;
	eMsgToGSToCSFromGC_CompCSGuideStepId = 8242; 
	//快速匹配消息
	eMsgToGSToCSFromGC_AskQuickBattle = 8250;
	eMsgToGSToCSFromGC_AskStopQuickBattle = 8251;
	//房间相关消息
	eMsgToGSToCSFromGC_AskCreateRoom = 8260;
	eMsgToGSToCSFromGC_AskAddRoom = 8261;
	eMsgToGSToCSFromGC_AskLeaveRoom = 8262;
	eMsgToGSToCSFromGC_AskChangeRoomSeat = 8263;
	eMsgToGSToCSFromGC_AskReadyRoom = 8264;
	eMsgToGSToCSFromGC_AskCancelRoom = 8265;
	eMsgToGSToCSFromGC_AskStartRoom = 8266;
	eMsgToGSToCSFromGC_AskChatInRoom = 8267;
	eMsgToGSToCSFromGC_AskReEnterRoom = 8268;
	eMsgToGSToCSFromGC_AskRoomList = 8269;
	eMsgToGSToCSFromGC_AskGoodscfg = 8270;
	//匹配相关消息
	eMsgToGSToCSFromGC_AskCreateMatchTeam = 8280;
	eMsgToGSToCSFromGC_AskAddMatchTeam = 8281;
	eMsgToGSToCSFromGC_AskRemoveMatchTeam = 8282;
	eMsgToGSToCSFromGC_AskInviteJoinMatchTeam = 8283;
	eMsgToGSToCSFromGC_AskStartMatch = 8284;
	eMsgToGSToCSFromGC_AskStopMatch = 8285;
	eMsgToGSToCSFromGC_AskChangeNickName = 8286;
	eMsgToGSToCSFromGC_RequestMatchTeamList = 8287;
	eMsgToGSToCSFromGC_NotifyOneMatchNeedOneRet = 8288;
	//任务系统
	eMsgToGSToCSFromGC_AskOneTaskRewards = 8290;
	//杂七杂八的请进来(不要污染其它组)
	eMsgToGSToCSFromGC_AskChangeheaderId = 8500;
	eMsgToGSToCSFromGC_CDKReq = 8505;
	//cs gm cmd
	eMsgToGSToCSFromGC_GmCmd = 8501;
	//log 8550-8560
	eMsgToGSToCSFromGC_UIEvent = 8550;

	//结束符
	eMsgToGSToCSFromGC_End = 9000;
}

message AskOneTaskRewards
{
	optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskOneTaskRewards]; 
	optional uint32 task_guid = 2;
}

message CDKReq
{
	optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_CDKReq]; 
	optional string cdkstr = 2;
}

//eMsgToGSToCSFromGC_UIEvent
message CurtUIEvent
{
	optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_UIEvent]; 
	message UIEvent
	{
		optional uint32 uiidx = 2;
		optional uint32 eventNum = 3;
	}
	repeated UIEvent  eventlist = 2;
}

message AskRecoinRune
{
	optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskRecoinRune];
	optional uint32 rune_id = 2;
	optional uint32 pay_type = 3;
	optional uint32 cost = 4;
}

message BlackListOnlineInfo
{
	optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskBlackListOnlineInfo];
}

//eMsgToGSToCSFromGC_GmCmd
message AddCSGMCmd
{
	optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_GmCmd];
	optional string gmcmd = 2;
}
message NotifyOneMatchNeedOneRet
{
	optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_NotifyOneMatchNeedOneRet];
	optional uint32 mapid = 2;
	optional uint32 fightid = 3;
	optional bool isAccept = 4;
}

message AskChangeheaderId
{
	optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskChangeheaderId];
	optional uint32 newheaderid = 2;
}

//eMsgToGSToCSFromGC_AskChangeNickName
message ChangeNickName
{
	optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskChangeNickName];
	optional string newnickname = 2;
}

message AskCreateMatchTeam
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskCreateMatchTeam];
optional uint32 matchtype = 2;
optional uint32 mapid = 3;
}
message AskAddMatchTeam
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskAddMatchTeam];
optional string friendsNickName = 2;
}
message AskRemoveMatchTeam
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskRemoveMatchTeam];
}
message AskInviteJoinMatchTeam
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskInviteJoinMatchTeam];
optional string friendsNickName = 2;
}
message AskStartMatch
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskStartMatch];
}
message AskStopMatch
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskStopMatch];
}
message RequestMatchTeamList
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_RequestMatchTeamList];
}
////////////////////////////////////////////////////////////////////////////////////
message AskGoodscfg
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskGoodscfg];
}
message AskCSCreateGuideBattle
{
	optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskGuideBattle];
	optional int32 mapid = 2;
	enum guidetype
	{
		  first = 1;
		  second = 2;
		  other = 3;
	}
	optional  guidetype ntype = 3;
}

message AskQuickBattle
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskQuickBattle];
optional int32 mapid = 2;
}

message AskStopQuickBattle
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskStopQuickBattle];
}

message AskCreateRoom
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskCreateRoom];
optional int32 mapid = 2;
optional string passwd = 3;
}

message AskAddRoom
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskAddRoom];
optional uint64 battleid = 2;
optional string passwd = 3;
}

message AskLeaveRoom
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskLeaveRoom];
}

message AskChangeRoomSeat
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskChangeRoomSeat];
optional int32 newpos = 2;
}

message AskReadyRoom
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskReadyRoom];
}

message AskCancelRoom
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskCancelRoom];
}

message AskStartRoom
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskStartRoom];
}

message AskReEnterRoom
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskReEnterRoom];
}

message AskChatInRoom
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskChatInRoom];
optional string chat = 2;
}

//eMsgToGSToCSFromGC_AskGetMailGift
message AskGetMailGift
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_AskGetMailGift];
	optional int32  mailId = 2; 
}

//eMsgToGSToCSFromGC_AskMailInfo
message AskMailInfo
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_AskMailInfo];
	optional int32  mailId = 2; 
}

message AskBuyGoods
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_AskBuyGoods];
	optional int32 commondityid = 2;
	optional int32 consumetype = 3;
	optional uint32 num = 4;
}
enum notice_type
{
	type_none = 0;
	type_1 = 1;
	type_2 = 2;
	type_3 = 3;
}

//eMsgToGSToCSFromGC_UserAskGetCLReward
message UserAskGetCLReward
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_UserAskGetCLReward];
}

//eMsgToGSToCSFromGC_Notice
message Notice
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_Notice];
	optional notice_type type = 2;
	optional string notice = 3; 
}

message AskCurtNotice
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_AskCurtNotice]; 
	optional notice_type type=2;
}

//已换GameCompleteBaseInfo
//eMsgToGSToCSFromGC_AskComleteUserInfo
message CompleteInfo
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskComleteUserInfo];
optional string nickname = 2;
optional int32 headid = 3;
optional int32 sex = 4;
}

//已换EmsgTocs_AskAddToFriendList
//协议名不一致eMsgToGSToCSFromGC_AskAddToFriendList->ToGSToCSFromGC_AskAddToSNSList
message AskAddToSNSList
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_AskAddToSNSList];
	optional string nickname = 2;
	optional int32 type= 3;
}

//eMsgToGSToCSFromGC_AskCanInviteFriends
message AskCanInviteFriends
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_AskCanInviteFriends];
}

//eMsgToGSToCSFromGC_AskAddToSNSListByID
message AskAddToSNSListByID
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_AskAddToSNSListByID];
	optional uint64 userid = 2;
	optional int32 type= 3;
}

//已换EmsgTocs_AskRemoveFromFriendList
//协议名不一致eMsgToGSToCSFromGC_AskRemoveFromFriendList->eMsgToGSToCSFromGC_AskRemoveFromSNSList
message AskRemoveFromSNSList
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_AskRemoveFromSNSList];
	optional uint64 guididx = 2;
	optional int32 type= 3;
}

//已换EmsgTocs_AskSendMsgToUser
//eMsgToGSToCSFromGC_AskSendMsgToUser
message AskSendMsgToUser
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_AskSendMsgToUser]; 
	optional uint64 guididx = 2;
	optional string contents = 3;
}

//已换EmsgTocs_AskInviteFriendsToBattle
//eMsgToGSToCSFromGC_AskInviteFriendsToBattle
message AskInviteFriendsToBattle
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_AskInviteFriendsToBattle]; 
	optional int32	battleid = 2;
	optional uint64 guididx = 3;
}

//已换EmsgTocs_FindFriendPlayer
//协议名不一致eMsgToGSToCSFromGC_AskSendFriendPersonInfo->eMsgToGSToCSFromGC_AskQueryUserByNickName
message AskQueryUserByNickName
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_AskQueryUserByNickName];
	optional string nickname = 2;
}

//已换EmsgTocs__GCReplyAddFriendRequst
//eMsgToGSToCSFromGC_GCReplyAddFriendRequst
message GCReplyAddFriendRequst
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_GCReplyAddFriendRequst];
	optional uint64 guididx = 2;
	optional int32 reply = 3;
}

//已换EmsgTocsAskFinishUIGuideTask
//协议名不一致eMsgToGSToCSFromGC_FinishUIGuideTask->eMsgToGSToCSFromGC_CompCSGuideStepId
message GuideCSStepComp
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_CompCSGuideStepId];
	optional int32 guidepart = 2;	 
	optional  bool bcomp = 3;		 
}

//已换EmsgTocs__GCReplyDownTime
//协议名不一致eMsgToGSToCSFromGC_GCReplyDownTime->eMsgToGSToCSFromGC_GCReplyInviteToBattle
message GCReplyInviteToBattle
{
	optional MsgNum msgnum = 1[default = eMsgToGSToCSFromGC_GCReplyInviteToBattle];
	optional string nickname = 2;
}

message Login
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskLogin];
optional int32 sdk = 2;
optional int32 platform = 3;
optional string name = 4;
optional string passwd = 5; 
optional string ipaddress = 6; 
optional string equimentid = 7; 
}

message UserGameInfo
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskUserGameInfo];
}

message BuyRunes
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_BUYRunes];
optional int32 typeid = 2;
}

message EuipRunes
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_EuipRunes];
optional uint32 runeid = 2;
optional int32 topos = 3;
}

message UnEuipRunes
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_UnloadRunes];
optional int32 page = 2;
optional int32 pos = 3;
}

message ComposeRunes
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_ComposeRunes];
repeated uint32 runesid = 2;
}

message ReconnectToGame
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskReconnectGame];
optional string name = 2;
optional string passwd = 3;
}
message ComGuideStepID
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskGuideStepId];
optional int32 type = 2;
optional int32 step = 3;
optional int32 state = 4;
}

message AskRoomList
{
optional MsgNum msgnum = 1 [default = eMsgToGSToCSFromGC_AskRoomList];
}
